synapse/handlers/event_auth.py

```diff
@@ -63,9 +63,23 @@ async def check_auth_rules_from_context(
             self._store, event, batched_auth_events
         )
         auth_event_ids = event.auth_event_ids()
-        auth_events_by_id = await self._store.get_events(auth_event_ids)
+        logger.info("auth events %s", auth_event_ids)
+
         if batched_auth_events:
-            auth_events_by_id.update(batched_auth_events)
+            logger.info("Batched auth events %s", list(batched_auth_events.keys()))
+            auth_events_by_id = dict(batched_auth_events)
+            if set(auth_event_ids) - set(batched_auth_events):
+                logger.info(
+                    "fetching %s", set(auth_event_ids) - set(batched_auth_events)
+                )
+                auth_events_by_id.update(
+                    await self._store.get_events(
+                        set(auth_event_ids) - set(batched_auth_events)
+                    )
+                )
```

Comment:
Same here:

```suggestion
            auth_events_by_id = dict(batched_auth_events)
            needed_auth_event_ids = set(auth_event_ids) - set(batched_auth_events)
            if needed_auth_event_ids:
                logger.info("fetching %s", needed_auth_event_ids)
                auth_events_by_id.update(
                    await self._store.get_events(needed_auth_event_ids)
```

(Are we sure we want that log at `info`? Is this sometimes a big list?)

Reply:
Now that you mention it I think the logging is not necessary in general, I've dropped it.